pkg/clusterrouter/proxy/proxy_server.go

```diff
@@ -227,6 +231,13 @@ func (r *RemoteService) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 		er.Error(rw, req, err)
 		return
 	}
+
+	if httpstream.IsUpgradeRequest(req) {
```

Comment:
@superseb @StrongMonkey both suggested this offline - going to add a check here so that it doesn't break normal setups without proxy if there's some issue with new proxy.  `isProxySet && IsUpgradeRequest `